addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -88,6 +90,49 @@ PaymentForm.include({
             );
         });
     },
+    _chargeExistingToken: function (ev, $checkedRadio) {
+        var self = this;
+        var json_url = this.$el.data('json-action');
+        var formData = this.getFormData(this.$el);
+        if (ev.type === 'submit') {
+            var button = $(ev.target).find('*[type="submit"]')[0]
+        } else {
+            var button = ev.target;
+        }
+        this.disableButton(button);
+        var stripe = Stripe(formData.stripe_publishable_key);
+        return this._chargeStripeToken(stripe, json_url, formData).guardedCatch(function (error) {
+            // if the rpc fails, pretty obvious
+            self.enableButton(button);
+            self.displayError(
+                _t('Unable to save card'),
+                _t("We were not able to process your payment. ") +
+                    self._parseError(error)
+            );
+        });
+    },
+    _chargeStripeToken: function(stripe, json_tx_url, formData) {
+        var tx_reference;
+        var self = this;
+        return this._rpc({
+            route: json_tx_url,
```

Comment:
we can get json_tx_url from this.$el.data('json-action') directly, so I don't think we should pass it from both the place.

Reply:
agreed